CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+# Contributing
+
+Implementing and maintaining a protocol stack is a lot of work, therefore any
+help is appreciated, from creating issues, to contributing documentation, fixing
+issues and adding new features.
+
+Please follow the best-practice contribution guidelines as mentioned below when
+submitting any changes.
+
+### Contributions License Agreement
+
+Before contributing, please read the [Contributions License Agreement](./CLA.md).
+Every contributor must sign this agreement before any code is merged into the
+repository.
+
+### Git
+
+Please, do your changes with the develop branch in your forked repo
+and send us your pull request from your own forked repo feature-branches
+or fix-branches to the develop branch of our repo!
+
+In the case that Commit rights were granted to Contributors the following basic ground-rules should be applied:
+
+* No --force pushes or modifying the Git history in any way.
+* Non-master branches ought to be used for ongoing work.
+* External API changes and significant modifications ought to be subject to an internal pull-request to solicit feedback from other contributors.
+* Internal pull-requests to solicit feedback are encouraged for any other non-trivial contribution but left to the discretion of the contributor.
```

Comment:
What do you mean by "internal pull-requests"?

Reply:
Internal pull reuest means that it is not coming from a fork, nbut from "your repo" (in fact from you or - if that is relevant potentially later - from other you allow write access to your repo here). Idea is to not commit into main directly "just because someone can" but ideally also use PRs (as you do right now)

Reply:
I forced approved PR to be able to merge into main.

My two cents: let's use automatic rules for everything we can and no need to mention it here in that case. The shortest is the file, the more likely people will actually read it...